### DISCUSSION TITLE
Why using createWebHistory refreshing the page while navigating from Sidebar?

### DISCUSSION THREAD

**Discussion by james-tanuwidjaja** [coreui/coreui-free-vue-admin-template#293] [Q&A] [answered]
Hi,

As the title said, I want to use `createWebHistory` instead of `createWebHashHistory` for the router. And when I change it to `createWebHistory` it refresh the page while navigating through Sidebar. Is it possible to fix this? If it is, where do I fix this?

Thank you in advance!

**Comment by ryandiginomad** [comment, marked as answer]
I've got the same problem here. The workaround I did is to modify AppSidebarNav.js onClick event:
`
onClick: (e) => {
e.preventDefault()
props.navigate()
},
`
But still I think it's just a workaround solution, because the same problem happened to CDropdownItem, CBreadcrumbItem

**Reply by pervej-fac** [reply]
Code released under [the MIT license](https://github.com/coreui/coreui-free-react-admin-template/blob/master/LICENSE). There is only one limitation you can't can’t re-distribute the CoreUI as stock. You can’t do this if you modify the CoreUI. In past we faced some problems with persons who tried to sell CoreUI based templates.

here is shown, we can not re-distribute. what is meant by this?

Suppose I have used the dashboard in my sales management software and others. and Can i sell the softwares? Have any restriction by this re-distribution issues?

**Comment by pervej-fac** [comment]
Code released under [the MIT license](https://github.com/coreui/coreui-free-react-admin-template/blob/master/LICENSE). There is only one limitation you can't can’t re-distribute the CoreUI as stock. You can’t do this if you modify the CoreUI. In past we faced some problems with persons who tried to sell CoreUI based templates.

here is shown, we can not re-distribute. what is meant by this?

Suppose I have used the dashboard in my sales management software and others. and Can i sell the softwares? Have any restriction by this re-distribution issues?
